src/orchestrator-process/src/lib.rs

```diff
@@ -499,13 +504,20 @@ impl NamespacedProcessOrchestrator {
             self.secrets_dir.display()
         ));
 
-        let scratch_directory = self.scratch_directory.as_ref().map(|dir| dir.join(id));
-        if let Some(scratch_directory) = &scratch_directory {
-            args.push(format!(
-                "--scratch-directory={}",
-                scratch_directory.display()
-            ));
-        }
+        let scratch_directory = if disk {
+            let scratch_directory = self.scratch_directory.as_ref().map(|dir| dir.join(id));
+            if let Some(scratch_directory) = &scratch_directory {
+                args.push(format!(
+                    "--scratch-directory={}",
+                    scratch_directory.display()
+                ));
+            } else {
+                // Ensured elsewhere
```

Comment:
Same comment as above, except the opposite direction. This should return "internal error: service requested disk but no scratch directory was configured".